props/prop_base.h

```diff
@@ -435,12 +416,68 @@ class PropBase : CommandParser, Looper, protected SaberBase {
 #else
     SaberBase::SetVariation(0);
 #endif
+  }
 
+  // Select preset (font/style)
+  virtual void SetPreset(int preset_num, bool announce) {
+    TRACE(PROP, "start");
+    bool on = BladeOff();
+    SaveColorChangeIfNeeded();
+    // First free all styles, then allocate new ones to avoid memory
+    // fragmentation.
+    UnsetCurrentStyles(preset_num);
+    if (announce) {
+      AnnouncePreset();
+    }
+    SetBladeLength();
+    SetBladeStyle();
     if (on) On();
     if (announce) SaberBase::DoNewFont();
     TRACE(PROP, "end");
   }
 
+  // Update Blade Style (no On/Off for use in Edit Mode)
+  void UpdateStyle(int preset_num) {
+    TRACE(PROP, "start");
+    SaveColorChangeIfNeeded();
+    // First free all styles, then allocate new ones to avoid memory
+    // fragmentation.
+    UnsetCurrentStyles(preset_num);
+    SetBladeLength();
+    SetBladeStyle();
+    TRACE(PROP, "end");
+  }
+
+    // Update Font & Style, skips Preon effect using FastOn (for use in Edit Mode and "fast" preset changes)
+  void UpdatePreset(int preset_num) {
```

Comment:
UpdatePreset is a terrible name. It's confusingly similar to SetPreset, but doesn't clarify what the difference is or when it should be used.

I think a function that is used in next_preset_fast() should probably be called SetPresetFast(int preset_num). I'm not sure if it makes sense to use that when just changing font though.